core/rawdb/accessors_chain.go

```diff
@@ -708,10 +708,12 @@ func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
 // DeriveLogFields fills the logs in receiptLogs with information such as block number, txhash, etc.
 func deriveLogFields(receipts []*receiptLogs, hash common.Hash, number uint64, txs types.Transactions) error {
 	logIndex := uint(0)
-	if len(txs) != len(receipts) {
+	// The receipts may include an additional "block finalization" receipt (only IBFT)
+	if !(len(txs) == len(receipts) || len(txs)+1 == len(receipts)) {
```

Comment:
Nit:
maybe rewrite it to:
`(len(txs) != len(receipts)) && (len(txs)+1 != len(receipts))`
it will look closer to the if clause we are replacing, and I feel like it's clearer (but probably subjective)